pkg/dockerregistry/server/pullthroughmanifestservice.go

```diff
@@ -0,0 +1,80 @@
+package server
+
+import (
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+// pullthroughManifestService wraps a distribution.ManifestService
+// repositories.
+type pullthroughManifestService struct {
+	distribution.ManifestService
+
+	repo                       *repository
+	pullFromInsecureRegistries bool
+}
+
+var _ distribution.ManifestService = &pullthroughManifestService{}
+
+func (m *pullthroughManifestService) Get(ctx context.Context, dgst digest.Digest, options ...distribution.ManifestServiceOption) (distribution.Manifest, error) {
+	manifest, err := m.ManifestService.Get(ctx, dgst, options...)
+	switch err.(type) {
+	case distribution.ErrManifestUnknownRevision:
+		break
+	case nil:
+		return manifest, nil
+	default:
+		context.GetLogger(ctx).Errorf("failed to get manifest using digest %q: %#v", dgst.String(), err)
```

Comment:
I don't see a need for logging this message twice (first time in manifestService).